### DISCUSSION TITLE
Parsing Result Not Expected

### DISCUSSION THREAD

**Discussion by tinygiant98** [tree-sitter/tree-sitter#1169] [Q&A] [answered]
I finally got the basic functionality of this parser working through a vscode extension and I'm going back to generate details and custom return types.  I added the following:
```
    nwnsc_macro: $ => seq(
      '__',
      token(choice(
        'FILE',
        'LINE',
        'COUNTER',
        'FUNCTION',
        'NSC_COMPILER_DATE',
        'NSC_COMPILER_TIME'),
      '__'
    )),
```
to be able to recognize specific macros that a one-off compiler recognizes.  But when I create a file with `__LINE__` in it, the parser recognizes the token as an `identifier` (and an error), not an `nwnsc_macro`.  There are no conflicts or other issues during the generate.  Any idea what I might be missing in this definition?  Seems all the other definitions I created worked just fine.  When I remove the `'__'` portions of the sequence, the tokens are recognized correctly.  Is there a more correct way to look for the double-underscore prior to and after the token?  Using `repeat` and `repeat1` also do not work.  I couldn't find anywhere else in the grammar where underscores would be recognized as any other token except in this rule and as an identifer (catch-all).

**Comment by ahlinc** [comment, marked as answer]
Try to assign **lexer** precedence for your token:
```js
    nwnsc_macro: $ => token(prec(2,
      seq(
        '__',
        choice(
          'FILE',
          'LINE',
          'COUNTER',
          'FUNCTION',
          'NSC_COMPILER_DATE',
          'NSC_COMPILER_TIME',
        ),
        '__'
      )
    )),
```

**Reply by tinygiant98** [reply]
Initially, this didn't work, but then it did.  Not sure why the first few attempts didn't, but I guess it doesn't really matter.  Oddly, I considered this yesterday and dismissed it, foolishly thinking that because there was no conflict during the generate that precedence was not required.  Thanks again for your time and expertise!
